#dynamixels: [
##shoulder_pan_controllerL', shoulder_tilt_controllerL',
#'elbow_pan_controllerL', 'elbow_tilt_controllerL',
#'wrist_tilt_controllerL', 'wrist_rotate_controllerL', 
#'finger_right_controllerL', finger_left_controllerL', 
##'shoulder_pan_controllerR', shoulder_tilt_controllerR', 
#'elbow_tilt_controllerR', elbow_pan_controllerR', 
#'wrist_tilt_controllerR', wrist_rotate_controllerR',
#'finger_right_controllerR', finger_left_controllerR', 
#'kinect_tilt_controller', kinect_pan_controller', laser_tilt_controller']
#left_arm_joints: [
##shoulder_pan_controllerL', shoulder_tilt_controllerL',
#'elbow_pan_controllerL', 'elbow_tilt_controllerL',
#'wrist_tilt_controllerL', 'wrist_rotate_controllerL', 
#'finger_right_controllerL', finger_left_controllerL'] 
#right_arm_joints: [
##'shoulder_pan_controllerR', shoulder_tilt_controllerR', 
#'elbow_tilt_controllerR', elbow_pan_controllerR', 
#'wrist_tilt_controllerR', wrist_rotate_controllerR',
#'finger_right_controllerR', finger_left_controllerR'] 

shoulder_pan_controllerL:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller_dual_motor
        type: JointPositionControllerDual
    joint_name: shoulder_pan_jointL
    joint_speed: .8
    motor_master:
        id: 2
        init: 512
        min: 0
        max: 1023
    motor_slave:
        id: 22

elbow_pan_controllerL:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: elbow_pan_jointL
    joint_speed: 0.8
    motor:
        id: 21
        # init: 512 - WRONG HW CONFIG?
        init: 256
        min:  440
        max:  1024

elbow_tilt_controllerL:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller_dual_motor
        type: JointPositionControllerDual
    joint_name: elbow_tilt_jointL
    joint_speed: 1.0
    motor_master:
        id: 16
        init: 192
        min: 226
        max: 1023
    motor_slave:
        id: 8

wrist_tilt_controllerL:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller_dual_motor
        type: JointPositionControllerDual
    joint_name: wrist_tilt_jointL
    joint_speed: 1.0
    motor_master:
        id: 9
        init: 512
        min: 226
        max: 798
    motor_slave:
        id: 11

wrist_rotate_controllerL:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: wrist_rotate_jointL
    joint_speed: 1.0
    motor:
        id: 6
        init: 500
        min: 0
        max: 1023

finger_right_controllerL:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: finger_right_jointL
    joint_speed: 1.0
    motor:
        id: 7
        init: 400
        min: 400
        max: 575

finger_left_controllerL:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: finger_left_jointL
    joint_speed: 1.0
    motor:
        id: 5
        init: 600
        min: 400
        max: 550

shoulder_pan_controllerR:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller_dual_motor
        type: JointPositionControllerDual
    joint_name: shoulder_pan_jointR
    joint_speed: .8
    motor_master:
        id: 17
        init: 512
        min: 0
        max: 1023
    motor_slave:
        id: 20

elbow_tilt_controllerR:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller_dual_motor
        type: JointPositionControllerDual
    joint_name: elbow_tilt_jointR
    joint_speed: 1.0
    motor_master:
        id: 13
        init: 192
        min: 226
        max: 1023
    motor_slave:
        id: 3

elbow_pan_controllerR:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: elbow_pan_jointR
    joint_speed: 0.8
    motor:
        id: 19
        init: 512
        min:  286
        max:  818


wrist_tilt_controllerR:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller_dual_motor
        type: JointPositionControllerDual
    joint_name: wrist_tilt_jointR
    joint_speed: 1.0
    motor_master:
        id: 4
        init: 512
        min: 226
        max: 798
    motor_slave:
        id: 14

wrist_rotate_controllerR:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: wrist_rotate_jointR
    joint_speed: 1.0
    motor:
        id: 18
        init: 505
        min: 205
        max: 740

finger_right_controllerR:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: finger_right_jointR
    joint_speed: 1.0
    motor:
        id: 15
        init: 600
        min: 400
        max: 600

finger_left_controllerR:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: finger_left_jointR
    joint_speed: 1.0
    motor:
        id: 10
        init: 400
        min: 400
        max: 600

kinect_tilt_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: kinect_tilt_joint
    joint_speed: 0.80
    motor:
        id: 23
        init: 512
        min: 300
        max: 800

kinect_pan_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: kinect_pan_joint
    joint_speed: 0.80
    motor:
        id: 1
        init: 512
        min: 0
        max: 1023

laser_tilt_controller:
    controller:
        package: dynamixel_controllers
        module: joint_position_controller
        type: JointPositionController
    joint_name: laser_tilt_joint
    joint_speed: 0.1
    motor:
        id: 12
        init: 512
        min: 212
        max: 812


